components/Masthead/Masthead.styles.ts

```diff
@@ -1,7 +1,13 @@
-export const root = 'w-full absolute top-0 z-50 transition-colors h-[7.6rem]';
+export const root = 'group w-full absolute top-0 z-50 transition-colors';
 
 // Use a wider centered container (1800px wide at 4XL (2000px) breakpoint)
 export const wrapper = 'cc 3xl:px-100 4xl:px-[calc((100%-1800px)/2)] py-12 sm:py-13 lg:py-20 transition';
 
 // Pass in a root style for the lock up to scale everything within proportionally
-export const lockup = 'sm:-mt-02em shrink-0 sm:text-17 md:text-22 lg:text-28';
+// We use the group-has- TW selector to make the lockup white when the main nav is open
+export const lockup = 'relative top-2 sm:top-0 z-[300] sm:-mt-02em shrink-0 sm:text-17 md:text-22 lg:text-28 group-has-[nav[data-headlessui-state="open"]]:*:*:text-white group-has-[nav[data-headlessui-state="open"]]:even:*:*:bg-white';
+
```

Comment:
This is how I use the has selector to change the logo color and the "Giving to Stanford" link color from black to white when the main nav panel is open on a light theme hero story page, e.g., Test/Test Stories/Yvonne Test New Story Hero. I've tested this TW utility group-has-, it has limited capabilities, e.g., `group-has-[data-headlessui-state="open"]`- on its own won't work, but how I have it here does.

Reply:
`group-has-[nav[data-headlessui-state="open"]]:even:*:*:bg-white` targets the 2nd element inside the logo lockup, which is the vertical bar.

Reply:
It's not super well documented on the TW site but it's from this section
https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-the-descendants-of-a-group

Reply:
Woof, that is a css class that would be better in vanilla CSS. That is a doozy.

Reply:
Does the headless ui give you access to the popover state through context or something? Might be easier to toggle a single `bg-white` on the element if you can know about the state of the component.

Reply:
I'm just amazed that it worked at all 🤣

Reply:
@sherakama Headless UI doesn't give an easy way to lift the state up to the parent Masthead so I chose to go with the has selector route. Also partly because they don't have any light pages, and has actually said they prefer the dark, so I chose the faster but imperfect method to deal with it here 😂